### PR TITLE
Checking current build configuration instead of tslint extended prompt

### PR DESCRIPTION
**Pull request by AJIXuMuK**
#### Category
- [x] New Feature
- [ ] New Framework
- [ ] Bugfix
- [ ] Documentation fixed
- Related issues: fixes #X, partially #Y, mentioned in #Z

#### What's in this Pull Request?

This PR contains changes for Vue.js generator and loading `fork-ts-checker-webpack-plugin` plugin.
Previously, Vue.js generator contained additional prompt "tslint: extended". If user selects "No", then `fork-ts-checker-webpack-plugin` has not been added to webpack. That led to disabled type checking in `.vue` files.
For example, the code below will be compiled correctly:
```typescript
let str: string = 'abc';
let digit: number = str;
```
It could potentially lead to the errors that are hard to fix.

The thing is that necessity of `fork-ts-checker-webpack-plugin` plugin is caused by ts-loader's option `transpileOnly: true`. 
And this option is needed to correctly bundle solution in release mode only (we need to disable creation of additional assets).
Saying that, we can check `build.getConfig().production` property and
* set `transpileOnly: true`
* add `fork-ts-checker-webpack-plugin` plugin to webpack.

In that case we won't lose performance while debugging and developing, and also will have type check enabled in .vue files.

I've tested the performance before this change with tslint extended: No and after.
The difference on two different projects for `gulp` was less than a second:
1. ~8.6s vs ~9.3s for OOTB web part with no changes
2. ~11s vs ~12s for web part with additional methods in `.vue` files with type errors in it.

### CODE REVIEW

**Comment by StfBauer** [comment]
SPFx already uses TSLint. If you think there is some configuration missing please focus on this linter.
Thank you. 👍

**Comment by StfBauer** [comment]
Yes but I do not want to have this in general not in the solution it would be better if TSLint does the syntactical code check. The Type check what you are referring too is not done by a "Linter" it is done by the TypeScript compiler.

**Comment by AJIXuMuK** [comment]
Just explain why do you think that the current approach with promtp “tslint extended” is better than the one that I have proposed.

**Comment by StfBauer** [comment]
Because I want the get completely rid of `ForkTsCheckerWebpackPlugin`.

**Comment by AJIXuMuK** [comment]
Do you know how and when?

**Comment by StfBauer** [comment]
Not yet but that is the challenge 😀 or if we still need to use it. How does the correct configuration look like because right now it looks a little bit off.

**Comment by AJIXuMuK** [comment]
I think there are 2 separate things here.
**First** 
TSLint doesn't check `.vue` files (syntactical errors). For example,
```typescript
const a: string[] = ["abc", "cde"];
    a.filter(a => {
      a.indexOf("a") !== -1;
    });
```
The code above will lead to the next output if added to `.ts` file:
```bash
Warning - tslint - src/webparts/helloWorld/HelloWorldWebPart.ts(24,14): error no-shadowed-variable: Shadowed name: 'a'
Warning - tslint - src/webparts/helloWorld/HelloWorldWebPart.ts(24,21): error no-unused-expression: unused expression, expected an assignment or function call
```
But if added to `.vue` - we won't get that warnings.

**Second**
Type checking.
Type checking is a part of TypeScript compiling task, not TSLint. 
For OOTB SPFx it is custom `tsc` task (`@microsoft/gulp-core-build-typescript`) that runs before `vue-loader` and customized webpack and at that moment doesn't process `.vue` files.

That's why we actually need `ts-loader` configuration for Vue files.
And we will need `ForkTsCheckerWebpackPlugin` when we're using `transpileOnly: true` in `ts-loader`.

Hopefully, later on, we will find a way to check syntactical errors in `.vue` using SPFx TSLint task. In that case we'll be able to remove `tslint: true` setting from the plugin. But not the plugin completely.

Saying all that, please look at the last commit.

It uses plugin for semantic errors AND sets `tslint` setting based on the prompt.
Basically, now we won't lose type checking and TSLint check will be optional.
TSLint part optional part could be removed later if we find a way to use TSLint gulp task from SPFx itself.

**Comment by StfBauer** [comment]
All understandable and valid points but during my research, I recognised may be the root cause. Until I get the whole picture of the problem this is just a patch up. The problem, in general, is the plugin loading right now in the SPFx build chain. Right now we have a working, if not an optimal solution and the user has the choice to choose between slower builds and extended linting or improved performance without extended linting.

We need to provide a solid working solution, not a quick fixup.

In addition, there are important parts missing to make a better performance. I am but always open for suggestions on the issue list.

I also experimented with cache and thread loader that should absolutely be part of the next version and is currently used by VUE CLI. This also influences the PR.

It would be great if you can have also a look at how the VueJS CLI handle things.
The first release is out. Next should be a fundamentally improved version.
